sdk/analytics/src/main/java/com/klaviyo/analytics/state/State.kt

```diff
@@ -0,0 +1,55 @@
+package com.klaviyo.analytics.state
+
+import com.klaviyo.analytics.model.Keyword
+import com.klaviyo.analytics.model.Profile
+import com.klaviyo.analytics.model.ProfileKey
+
+typealias StateObserver = (key: Keyword?) -> Unit
+
+interface State {
+    var externalId: String?
+    var email: String?
+    var phoneNumber: String?
+    val anonymousId: String?
+    var pushToken: String?
+    var pushState: String?
+
+    /**
+     * Register an observer to be notified when state changes
+     *
+     * @param observer
+     */
+    fun onStateChange(observer: StateObserver)
+
+    /**
+     * De-register an observer from [onStateChange]
+     *
+     * @param observer
+     */
+    fun offStateChange(observer: StateObserver)
+
+    /**
+     * Get all user data in state as a [Profile] model object
+     */
+    fun get(withAttributes: Boolean = false): Profile
+
+    /**
+     * Update user state from a new [Profile] model object
+     */
+    fun set(profile: Profile)
+
+    /**
+     * Set an individual attribute
+     */
+    fun set(key: ProfileKey, value: String)
```

Comment:
hmm reading around, i feel that these names can be a little clearer, `getProfile`, `setProfile`, `setAttribute`

Reply:
fair, this is swift-y, but without the argument label enforcement of swift

Reply:
when reading within the `State` class, when I see `get()` its a bit of processing to remember that `get()` returns a `Profile` otherwise initially i would expect it to return a copy of `State`

Reply:
Yeah it used to be `UserInfo.getAsProfile`, I'm going to go back to that.